src/main/java/com/commercetools/sync/states/utils/StateReferenceResolutionUtils.java

```diff
@@ -42,22 +42,24 @@
    *   </tbody>
    * </table>
    *
-   * <p><b>Note:</b> The transition references should be expanded with a key. Any reference that is
-   * not expanded will have its id in place and not replaced by the key will be considered as
-   * existing resources on the target commercetools project and the library will issues an
-   * update/create API request without reference resolution.
+   * <p><b>Note:</b> The transition references should be cached(idToKey value fetched and stored in
+   * a map). Any reference that is not cached will have its id in place and not replaced by the key
+   * will be considered as existing resources on the target commercetools project and the library
```

Comment:
```suggestion
   * a map). Any reference, which have its id in place and not replaced by the key, it would not
   * be found in the map. In this case, this reference will be considered as existing resources  on the target commercetools project and the library
```
 
   * b